extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -118,21 +111,54 @@ export class AmpAdApiHandler {
           }
         }, this.is3p_, this.is3p_));
 
+    // Install API that listen to ad response
+    const renderStartImplemented =
+        (waitForRenderStart.indexOf(this.baseInstance_.adType) >= 0);
+
+    if (renderStartImplemented) {
+      // If support render-start, create a race between render-start no-content
+      this.adResponsePromise_ = listenForOncePromise(this.iframe_,
+        ['render-start', 'no-content'], this.is3p_).then(info => {
+          if (info.data.type == 'render-start') {
```

Comment:
https://github.com/ampproject/amphtml/blob/master/src/iframe-helper.js#L212

I'll say yes. Here `data`must have a `data.type` to get the correct listener